riscv/zicfiss.h

```diff
@@ -0,0 +1,29 @@
+#ifndef _RISCV_ZICFISS_H
+#define _RISCV_ZICFISS_H
+
+#define SET_XSSE(xSSE) \
+  reg_t prv = STATE.prv; \
+  if (prv == PRV_M) \
+    xSSE = false; \
+  else if (prv == PRV_S || prv == PRV_HS) \
+    xSSE = (get_field(STATE.menvcfg->read(), MENVCFG_SSE) == 1); \
+  else if (p->extension_enabled('S')) \
+    xSSE = (get_field(STATE.senvcfg->read(), SENVCFG_SSE) == 1); \
+  if (prv == PRV_S && STATE.v) \
+    xSSE = xSSE & (get_field(STATE.henvcfg->read(), HENVCFG_SSE) == 1);
+
```

Comment:
How about this?
```
#define xSSE()\
  ((prv != M) && get_field(STATE.menvcfg->read(), MENVCFG_SSE) && p->extension_enabled('S') &&
   ((STATE.v && get_field(STATE.henvcfg->read(), HENVCFG_SSE)) || !STATE.v) &&
   (((prv == PRV_U) && get_field(STATE.senvcfg->read(), SENVCFG_SSE)) || (prv != PRV_U)))
```
and then we can use this directly:
```
if (xSSE()) {
...
}
```

Reply:
Thanks, your code is a more concise.
```
#define xSSE() \
  ((STATE.prv != PRV_M) && get_field(STATE.menvcfg->read(), MENVCFG_SSE) && \
  p->extension_enabled('S') && \
  ((STATE.v && get_field(STATE.henvcfg->read(), HENVCFG_SSE)) || !STATE.v) && \
  (((STATE.prv == PRV_U) && get_field(STATE.senvcfg->read(), SENVCFG_SSE)) || (STATE.prv != PRV_U)))
```